lib/Segments/DynamicSegments/Filters/WooCommerceSubscription.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace MailPoet\Segments\DynamicSegments\Filters;
+
+use MailPoet\Entities\DynamicSegmentFilterEntity;
+use MailPoet\Entities\SubscriberEntity;
+use MailPoetVendor\Doctrine\DBAL\Query\QueryBuilder;
+use MailPoetVendor\Doctrine\ORM\EntityManager;
+
+class WooCommerceSubscription implements Filter {
+  const ACTION_HAS_ACTIVE = 'hasActiveSubscription';
+
+  /** @var EntityManager */
+  private $entityManager;
+
+  public function __construct(EntityManager $entityManager) {
+    $this->entityManager = $entityManager;
+  }
+
+  public function apply(QueryBuilder $queryBuilder, DynamicSegmentFilterEntity $filter): QueryBuilder {
+    global $wpdb;
+    $filterData = $filter->getFilterData();
+    $productId = (int)$filterData->getParam('product_id');
+    $subscribersTable = $this->entityManager->getClassMetadata(SubscriberEntity::class)->getTableName();
+    return $queryBuilder->innerJoin(
```

Comment:
I unified the code formatting. As for the alias, we have to use the subscribers' table since the $queryBuilder that is passed as an argument is expected to select from the subscribers' table.